pickfirst balancer can cause a nil pointer deref panic due to data race

### What version of gRPC are you using?
master (878cea23105641ab36ceb688af326139cf60b842)

### What version of Go are you using (`go version`)?
`go version go1.17 linux/amd64`

### What operating system (Linux, Windows, …) and version?
`Ubuntu 16.04.7 LTS`

### What did you do?
By inserting couple of `time.Sleep` calls inside grpc code to force a certain execution order, I can repro a nil pointer deref panic that is happening in production for us. Please see here for a repro test: https://github.com/grpc/grpc-go/compare/master...entombedvirus:pickfirst-panic-demo?expand=1

### What did you expect to see?
pickfirst balancer should not panic.

### What did you see instead?
panics.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x7f974c]

goroutine 22 [running]:
google.golang.org/grpc.(*pickfirstBalancer).ExitIdle(0xc00027e7b0)
        /home/rravi/work/grpc-go/pickfirst.go:131 +0x6c
google.golang.org/grpc.(*ccBalancerWrapper).watcher(0xc000180730)
        /home/rravi/work/grpc-go/balancer_conn_wrappers.go:102 +0x3cd
created by google.golang.org/grpc.newCCBalancerWrapper
        /home/rravi/work/grpc-go/balancer_conn_wrappers.go:67 +0x246
```

### Notes
The simplest way to fix this  seems to be check if b.sc is nil before calling b.sc.Connect here: https://github.com/grpc/grpc-go/blob/master/pickfirst.go#L128-L130 but wasn't sure if the panic is symptomatic of a more fundamental issue re: how state is managed.

Please let me know if you would like me to open a PR with that fix / anything else or if grpc team want to implement the fix internally.

--- a/pickfirst.go
+++ b/pickfirst.go
@@ -125,7 +125,7 @@ func (b *pickfirstBalancer) Close() {
 }
 
 func (b *pickfirstBalancer) ExitIdle() {
-	if b.state == connectivity.Idle {
+	if b.sc != nil && b.state == connectivity.Idle {
 		b.sc.Connect()
 	}
 }

